icu4c/source/common/unicode/unistr.h

```diff
@@ -327,6 +328,53 @@ class U_COMMON_API UnicodeString : public Replaceable
    */
   inline bool operator== (const UnicodeString& text) const;
 
+#ifndef U_HIDE_DRAFT_API
+  /**
+   * Equality operator. Performs only bitwise comparison with a
+   * std::u16string_view or (if U_SIZEOF_WCHAR_T==2) std::wstring_view.
+   * TODO: Should we document this as what it is, "convertible to one of these string views",
+   * rather than claiming that it needs to *be* one?
+   *
+   * @param text The string view to compare to this string.
+   * @return true if `text` contains the same characters as this one, false otherwise.
+   * @draft ICU 76
+   */
+  template<typename StringView, typename = std::enable_if_t<ConvertibleToU16StringView<StringView>>>
+  // TODO: I am tempted to stick this template line into a #define macro like
+  // U_TEMPLATE_StringView_IF_CONVERTIBLE_TO_U16STRING_VIEW
```

Comment:
WDYT?

Reply:
I don’t think a macro is helpful. With a dedicated named trait this is very concise and idiomatic C++17, whereas a macro is always a puzzle.
(In C++20 it would get more compact, as discussed in another place.)

(Also I find a macro that declares a specific magic name to be very distasteful, though you could parameterize it on the type name. But my earlier point still stands.)